ts/Data/Formula/FormulaProcessor.ts

```diff
@@ -0,0 +1,470 @@
+/* *
+ *
+ *  (c) 2009-2023 Highsoft AS
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ *  Authors:
+ *  - Sophie Bremer
+ *
+ * */
+
+
+'use strict';
+
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+
+import type DataTable from '../DataTable';
+import type {
+    Arguments,
+    Formula,
+    Function,
+    Item,
+    Operator,
+    Range,
+    Reference,
+    Term,
+    Value
+} from './FormulaTypes';
+
+
+import FormulaTypes from './FormulaTypes.js';
+const {
+    asNumber,
+    isFormula,
+    isFunction,
+    isOperator,
+    isRange,
+    isReference,
+    isValue
+} = FormulaTypes;
+
+
+/* *
+ *
+ *  Declarations
+ *
+ * */
+
+
+/**
+ * Formula processor might not process a term.
+ * @private
+ */
+export interface FormulaProcessError extends Error {
+    message: string;
+    name: 'FormulaProcessError';
+}
+
+
+export interface ProcessorFunction {
+    (args: Arguments, table?: DataTable): (Value|Array<Value>);
+}
+
+
+/* *
+ *
+ *  Constants
+ *
+ * */
+
+
+const processorFunctions: Record<string, ProcessorFunction> = {};
+
+
+const processorFunctionNameRegExp = /^[A-Z][A-Z\.]*$/;
+
+
+/* *
+ *
+ *  Functions
+ *
+ * */
+
+
+/**
+ * Process a basic operation of two given values.
+ *
+ * @private
+ *
+ * @param {Highcharts.FormulaOperator} operator
+ * Operator between values.
+ *
+ * @param {Highcharts.FormulaValue} x
+ * First value for operation.
+ *
+ * @param {Highcharts.FormulaValue} y
+ * Second value for operation.
+ *
+ * @return {Highcharts.FormulaValue}
+ * Operation result. `NaN` if operation is not support.
+ */
+function basicOperation(
+    operator: Operator,
+    x: Value,
+    y: Value
+): Value {
+
+    switch (operator) {
+        case '=':
+            return x == y ? 1 : 0; // eslint-disable-line eqeqeq
+        case '<':
+            return x < y ? 1 : 0;
```

Comment:
You can also use Number(x < y)

Reply:
These need to be booleans. So this needs a different form of fix.